staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/managedfieldsupdater_test.go

```diff
@@ -29,42 +30,255 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
-func TestNoManagedFieldsUpdateDoesntUpdateTime(t *testing.T) {
-	f := NewTestFieldManager(schema.FromAPIVersionAndKind("v1", "Pod"), "", nil)
+func TestManagedFieldsUpdateDoesModifyTime(t *testing.T) {
+	f := NewTestFieldManager(schema.FromAPIVersionAndKind("v1", "ConfigMap"), "", nil)
 
-	obj := &unstructured.Unstructured{Object: map[string]interface{}{}}
-	if err := yaml.Unmarshal([]byte(`{
+	updateObject(t, &f, "fieldmanager_test", []byte(`{
 		"apiVersion": "v1",
-		"kind": "Pod",
+		"kind": "ConfigMap",
 		"metadata": {
-			"name": "pod",
+			"name": "configmap",
 			"labels": {"app": "nginx"}
 		},
-	}`), &obj.Object); err != nil {
-		t.Fatalf("error decoding YAML: %v", err)
+	}`))
+	previousManagedFields := f.ManagedFields()
+
+	time.Sleep(time.Second)
+
+	updateObject(t, &f, "fieldmanager_test", []byte(`{
+		"apiVersion": "v1",
+		"kind": "ConfigMap",
+		"metadata": {
+			"name": "configmap",
+			"labels": {"app": "nginx2"}
+		},
+	}`))
+	newManagedFields := f.ManagedFields()
+
+	if reflect.DeepEqual(previousManagedFields, newManagedFields) {
+		t.Errorf("ManagedFields time has not been updated:\n%v", newManagedFields)
```

Comment:
That's not really what you're comparing so that error might be misleading (also applies below)

Reply:
I am not sure to understand it, should it directly compare the Time field of both entries?
Current, if the entries are identical, that means the time has not been updated.

Reply:
I added a test case named `TestTakingOverManagedFieldsDuringUpdateDoesNotModifyPreviousManagerTime` for the Update.

Reply:
You're comparing `ManagedFields`, but the error reads `time has not been updated`, which is slightly misleading.

Reply:
But the code looks good now, thanks